packages/interchain-token-service-api/src/abi.rs

```diff
@@ -0,0 +1,427 @@
+use alloy_primitives::{Bytes, FixedBytes, U256};
+use alloy_sol_types::{sol, SolValue};
+use cosmwasm_std::{HexBinary, Uint256};
+use error_stack::Report;
+
+use crate::{
+    error::Error,
+    primitives::{ITSMessage, ITSRoutedMessage},
+    TokenId, TokenManagerType,
+};
+
+sol! {
+    enum MessageType {
+        InterchainTransfer,
+        DeployInterchainToken,
+        DeployTokenManager,
+        RoutedCall,
+    }
+
+    struct InterchainTransfer {
+        uint256 messageType;
+        bytes32 tokenId;
+        bytes sourceAddress;
+        bytes destinationAddress;
+        uint256 amount;
+        bytes data;
+    }
+
+    struct DeployInterchainToken {
+        uint256 messageType;
+        bytes32 tokenId;
+        string name;
+        string symbol;
+        uint8 decimals;
+        bytes minter;
+    }
+
+    struct DeployTokenManager {
+        uint256 messageType;
+        bytes32 tokenId;
+        uint256 tokenManagerType;
+        bytes params;
+    }
+
+    struct RoutedCall {
+        uint256 messageType;
+        string chain;
```

Comment:
Is this the destination chain or the source chain?

Reply:
I'll change the name to remote_chain and add a comment. It's used for both. When ITS edge contract sends a RoutedCall, the remote_chain is the true destination, and when it receives the GMP call from ITS hub it's the true source_chain.